### ISSUE TITLE
Cannot download mobilenet_v2 pretrained weights - 403 Forbidden error

### ISSUE THREAD

**Issue by C-Achard** [DeepLabCut/DeepLabCut#3243]
### Discussed in https://github.com/DeepLabCut/DeepLabCut-live/discussions/172

<div type='discussions-op-text'>

<sup>Originally posted by **Spoorthy-Siddannaiah** March 13, 2026</sup>
<sup>Reposted & moved from DeepLabCut-live. Reason : issue relevant to DLC, not DLC-live</sup>

---

  Hi,
                                                                                                                                                                                                        
  I am using DeepLabCut 2.3 with mobilenet_v2_1.0 backbone on Windows 10                                                                                                                                
  with TensorFlow 2.10.                                                                                                                                                                                 
                                                                                                                                                                                                        
  When I try to train the network I get this error:                                                                                                                                                     
  NOT_FOUND: NewRandomAccessFile failed - mobilenet_v2_1.0_224.ckpt.data-00000-of-00001
  The system cannot find the file specified.                                                                                                                                                            
                  
  The file mobilenet_v2_1.0_224.ckpt.data-00000-of-00001 is missing                                                                                                                                     
  from my pretrained folder.
                                                                                                                                                                                                        
  I tried downloading from:
  https://storage.googleapis.com/mobilenet_v2/checkpoints/mobilenet_v2_1.0_224.tgz
  But getting HTTP Error 403: Forbidden from multiple networks and also                                                                                                                                 
  from Google Colab.                                                                                                                                                                                    
                                                                                                                                                                                                        
  I also tried generating the checkpoint using Keras MobileNetV2 with                                                                                                                                   
  imagenet weights in Colab and saving as .ckpt, but then I get:
  NOT_FOUND: Key MobilenetV2/Conv/BatchNorm/beta not found in checkpoint                                                                                                                                
                  
  This means the Keras-saved weights have different variable names than                                                                                                                                 
  what DLC expects (TF1 slim format).
                                                                                                                                                                                                        
  Running check_for_weights() and download.sh also give 403 error.                                                                                                                                      
                                                                                                                                                                                                        
  Could someone please share the correct mobilenet_v2_1.0_224 checkpoint                                                                                                                                
  files in TF1 slim format, or provide a working alternative download link?

  Thank you
</div>

**Comment by C-Achard** [comment]
Hello @Spoorthy-Siddannaiah,

This seems to be a current issue with accessing pre-trained weights for several models in tensorflow/models - see https://github.com/tensorflow/models/issues/13610 or https://github.com/tensorflow/models/issues/13561.

I've checked personally and I cannot access any of their weights at the time either, I'm afraid this isn't in our hands currently.

If you would like help with using the PyTorch version (if possible for your use case), please let me know. Otherwise, we can try and find alternatives suitable for your project.

Best,
Cyril

**Comment by C-Achard** [comment]
Related : https://github.com/DeepLabCut/DeepLabCut/issues/2937 and a [potential workaround](https://github.com/DeepLabCut/DeepLabCut/issues/2937#issuecomment-2832923859) for obtaining those weights

**Comment by Spoorthy-Siddannaiah** [comment]
@C-Achard 

Thank you for the response!                                                                                                                                                                                                                                                
                                                                                                                                                                                                                                                                             
  Regarding PyTorch version - would switching require a complete                                                                                                                                                                                                             
  reinstallation or can it work alongside my current TensorFlow 2.10                                                                                                                                                                                                         
  setup with DLC 2.3?                                                                                                                                                                                                                                                        
                  
  I would also like to explore alternatives for getting the mobilenet                                                                                                                                                                                                        
  weights. I tried the following approach in Google Colab:
                                                                                                                                                                                                                                                                             
  1. Loaded MobileNetV2 from Keras with imagenet weights                                                                                                                                                                                                                     
  2. Saved as TF1 checkpoint                                                                                                                                                                                                                                                 
  3. Copied the files to the pretrained folder                                                                                                                                                                                                                               
                                                                                                                                                                                                                                                                             
  But got this error when training:
  NOT_FOUND: Key MobilenetV2/Conv/BatchNorm/beta not found in checkpoint                                                                                                                                                                                                     
                                                                                                                                                                                                                                                                             
  This means the Keras-saved weights have different variable names                                                                                                                                                                                                           
  than what DLC expects in TF1 slim format.                                                                                                                                                                                                                                  
                                                                                                                                                                                                                                                                             
  Is there a way to:                                                                                                                                                                                                                                                         
  1. Convert Keras MobileNetV2 weights to TF1 slim format with
     correct variable names?                                                                                                                                                                                                                                                 
  2. Or any other alternative to get the correct pretrained weights?

**Comment by C-Achard** [comment]
Hello @Spoorthy-Siddannaiah,

Yes, switching to PyTorch safely would entail a clean install. It is possible, but not really recommended, to have both TF and PT coexist in the same environment,; in general this is best avoided.

For the weights, conversion is not straightforward as you'd require a custom variable-name mapping as you mention. DLC explicitly requires weights in the TF-Slim format, not a Keras-exported checkpoint. 

Have you tried the listed workaround above for downloading the weights ? That should be the most reliable option, since they correspond to the same MobileNetV2 checkpoints that DeepLabCut explicitly references in its TensorFlow pretrained-model URL map:
> Related : [#2937](https://github.com/DeepLabCut/DeepLabCut/issues/2937) and a [potential workaround](https://github.com/DeepLabCut/DeepLabCut/issues/2937#issuecomment-2832923859) for obtaining those weights

Let me know if this helps you retrieve and use the weights.

Best,
Cyril

**Comment by Spoorthy-Siddannaiah** [comment]
@C-Achard 

Hi Cyril,                                                                                                                                                                                                                                                                  
                                                                                                                                                                                                                                                                             
  Thank you for the clarification regarding PyTorch and the TF-Slim format explanation!                                                                                                                                                                                      
                                                                                                                                                                                                                                                                             
  I tried the Wayback Machine workaround but unfortunately it is not working -                                                                                                                                                                                               
  all mobilenet URLs only show orange (404/403) in the Wayback Machine calendar.                                                                                                                                                                                                              
                                                                                                                                                                                                                                                                             
  Would it be possible for you or anyone from the team to share the                                                                                                                                                                                                          
  mobilenet_v2_1.0_224 checkpoint files directly?                                                                                                                                                                                                                            
  Even via Google Drive or any file sharing service would be fine.                                                                                                                                                                                                           
                                                                                                                                                                                                                                                                             
  I specifically need these 3 files in TF-Slim format:                                                                                                                                                                                                                       
  - mobilenet_v2_1.0_224.ckpt.index                                                                                                                                                                                                                                          
  - mobilenet_v2_1.0_224.ckpt.meta                                                                                                                                                                                                                                           
  - mobilenet_v2_1.0_224.ckpt.data-00000-of-00001                                                                                                                                                                                                                            
                                                                                                                                                                                                                                                                             
  Thank you so much for your help!

**Comment by C-Achard** [comment]
Okay, sorry to hear that. There was an internal discussion about hosting these weights ourselves, I will ask what the status is and whether we still have them somewhere.

https://github.com/DeepLabCut/DeepLabCut/issues/2937#issuecomment-2862426731

**Comment by deruyter92** [comment]
They can be temporarily downloaded here, while we are working on a solution:
https://drive.google.com/drive/folders/1Rv9ARXi8OROvB2SSTc74HPC6NGzcPTta?usp=sharing

**Comment by C-Achard** [comment]
Hello @Spoorthy-Siddannaiah,

We hope the weights we shared were useful. I will go ahead and close this for now, feel free to reopen if needed.

Thanks !
